packages/ag-charts-enterprise/src/features/annotations/annotations.ts

```diff
@@ -514,15 +541,55 @@ export class Annotations extends _ModuleSupport.BaseModuleInstance implements _M
         );
     }
 
-    private translateNode({
-        node,
-        datum,
-        translation,
-    }: {
-        node: AnnotationScene;
-        datum: AnnotationProperties;
-        translation: _ModuleSupport.Vec2;
-    }): AnnotationProperties | undefined {
+    /**
+     * Append an annotation datum to `this.annotationData`, applying default styles. This method is called when a user
+     * interacts with the chart to draw their own annotations.
+     */
+    private createAnnotation(type: AnnotationType, datum: AnnotationProperties, applyDefaults: boolean = true) {
+        this.annotationData.push(datum);
+
+        if (applyDefaults) {
+            const styles = this.ctx.annotationManager.getAnnotationTypeStyles(type);
+            if (styles) datum.set(styles);
+            this.defaults.applyDefaults(datum);
+        }
+
+        this.injectDatumDependencies(datum);
+        this.resetToolbarButtonStates();
+
+        this.removeAmbientKeyboardListener?.();
+        this.removeAmbientKeyboardListener = undefined;
+
+        this.update();
+    }
+
+    private injectDatumDependencies(datum: AnnotationProperties) {
+        if ('setLocaleManager' in datum) {
+            datum.setLocaleManager(this.ctx.localeManager);
+        }
+
+        if ('getVolume' in datum) {
+            datum.getVolume = this.getDatumRangeVolume.bind(this);
+        }
+    }
+
+    private getDatumRangeVolume(from: Point['x'], to: Point['x']) {
+        if (!isValidDate(from) || !isValidDate(to) || !this.processedData) return 0;
+
+        return this.processedData.data.reduce((sum, datum) => {
+            const key = datum.keys[0];
+            if (isValidDate(key) && key >= from && key <= to) {
+                return sum + datum.values[0];
+            }
+            return sum;
+        }, 0);
+    }
```

Comment:
This will reduce the data on every change to the measurement. It could be memoised, but I wasn't sure if there was a better overall approach anyway?

Reply:
This will work, but I wonder if it would be better to delegate to the relevant `Scale` to answer the question of what the distance is?